addons/token-exchange/agent_drpolicy_controller.go

```diff
@@ -0,0 +1,328 @@
+package addons
+
+import (
+	"context"
+	"fmt"
+	replicationv1alpha1 "github.com/csi-addons/kubernetes-csi-addons/apis/replication.storage/v1alpha1"
+	ramenv1alpha1 "github.com/ramendr/ramen/api/v1alpha1"
+	multiclusterv1alpha1 "github.com/red-hat-storage/odf-multicluster-orchestrator/api/v1alpha1"
+	"github.com/red-hat-storage/odf-multicluster-orchestrator/controllers/utils"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sort"
+	"time"
+)
+
+type DRPolicyReconciler struct {
```

Comment:
Since it's watching for a resource in hub cluster, we should run this controller in hub cluster. Otherwise we'll have the overhead of frequent cross cluster communication.

According to initial discussion, this controller should run on hub and create resources on ManagedCluster using ACM's ManifestWork.

Reply:
Looking into this

Reply:
done